pkg/cri/sbserver/container_create.go

```diff
@@ -161,8 +162,25 @@ func (c *criService) CreateContainer(ctx context.Context, r *runtime.CreateConta
 	}
 	log.G(ctx).Debugf("Use OCI runtime %+v for sandbox %q and container %q", ociRuntime, sandboxID, id)
 
-	spec, err := c.containerSpec(id, sandboxID, sandboxPid, sandbox.NetNSPath, containerName, containerdImage.Name(), config, sandboxConfig,
-		&image.ImageSpec.Config, append(mounts, volumeMounts...), ociRuntime)
+	platform, err := controller.Platform(ctx, sandboxID)
```

Comment:
For runtimes that support multiple guests (Windows prime example I guess) how do they decide what to return? This would be locked in at RunPodSandbox time, and the way we even went about that (on Windows) was to use a shim specific field that a user would need to set to tell us to fill in *both* the windows and linux portions of the spec: https://github.com/microsoft/hcsshim/blob/6547959343d65e0cf431dd9d589f45ee2b98172c/cmd/containerd-shim-runhcs-v1/options/runhcs.proto#L35

Spec generation for containers in the pod seems great here, but [in my issue](https://github.com/containerd/containerd/issues/7910) I was honestly thinking about the first problem of how can we tell the runtime what guest we want to launch at all first; containers to follow in this sandbox come for free pretty much after this which is great. Kata hasn't had to worry about this as it's always Linux (for now), Windows does, and I assume Darwin might to a degree.

Reply:
> For runtimes that support multiple guests (Windows prime example I guess) how do they decide what to return? 

With Sandbox API, you can launch multiple sandboxes with different OSes (depending on runtime class, annotation, etc). Here we ask platform per sandbox, not per runtime. So each sandbox instance will return target platform it's currently using. Hence each `CRI.CreateContainer` call will produce proper OCI spec for target OS.

Right, this one is focused on generating proper OCI spec, not how to tell runtime which OS to bootstrap.

Reply:
Perfect okay, I'll try and figure out the "tell shim what guest to launch bit". Ideally k8s could send something, but a config field here might suffice